src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -763,14 +755,59 @@ int wm_vuldet_send_agent_report(sqlite3 *db, OSHash *cve_table, agent_software *
+                        const char* middle = "and is";
+                        const char* condition2 = (pkg->nvd_cond->end_operation == END_INCLUDED) ? "less or equal than" : "less than";
+                        size_t size = strlen(start) + 1 + strlen(condition1) + 1 + strlen(pkg->nvd_cond->vuln_version) + 1 + 
+                                      strlen(middle) + 1 + strlen(condition2) + 1 + strlen(pkg->nvd_cond->fix_version);
+                        os_calloc(size + 1, sizeof(char), report->condition);
+                        sprintf(report->condition, "%s %s %s %s %s %s", 
+                        start, condition1, pkg->nvd_cond->vuln_version, 
+                        middle, condition2, pkg->nvd_cond->fix_version);
+                    }
+                    else if (pkg->nvd_cond->vuln_version) {
+                        const char* start = "package is";
+                        const char* condition = (pkg->nvd_cond->start_operation == START_INCLUDED) ? "greater or equal than" : "greater than";
+                        size_t size = strlen(start) + 1 + strlen(condition) + 1 + strlen(pkg->nvd_cond->vuln_version);
+                        os_calloc(size + 1, sizeof(char), report->condition);
+                        sprintf(report->condition, "%s %s %s", 
+                        start, condition, pkg->nvd_cond->vuln_version);
+                    }
+                    else if (pkg->nvd_cond->fix_version) {
+                        const char* start = "package is";
+                        const char* condition = (pkg->nvd_cond->end_operation == END_INCLUDED) ? "less or equal than" : "less than";
+                        size_t size = strlen(start) + 1 + strlen(condition) + 1 + strlen(pkg->nvd_cond->fix_version);
+                        os_calloc(size + 1, sizeof(char), report->condition);
+                        sprintf(report->condition, "%s %s %s", 
+                        start, condition, pkg->nvd_cond->fix_version);
+                    }
+                    else {
+                        const char* text = "reports as vulnerable all the versions of this package";
+                        size_t size = strlen(cve) + 1 + strlen(text);
+                        os_calloc(size + 1, sizeof(char), report->condition);
+                        sprintf(report->condition, "%s %s", cve, text);
+                    }
+                }
+
                 if (sqlite3_prepare_v2(db, vu_queries[VU_GET_CVE_INFO_FILTER_CVE], -1, &stmt, NULL) != SQLITE_OK) {
                     goto error;
                 }
                 sqlite3_bind_text(stmt, 1, cve, -1, NULL);
 
+                int cve_table_id = 0;
                 int result = sqlite3_step(stmt);
                 if (result == SQLITE_ROW) {
-                    cve_table_id = (char *)sqlite3_column_text(stmt, 0);
+                    cve_table_id = atoi((char *)sqlite3_column_text(stmt, 0));
```

Comment:
The call to `sqlite3_finalize` at line 828 was causing this pointer to be dereferenced. As a result, all the subsequent `atoi` calls, sometimes were returning a bad ID for the CVE being evaluated. This caused the loss of the fields `cvss`, `severity` and `references` from the alerts.